transfer/main.tf

```diff
@@ -0,0 +1,28 @@
+variable "name" {
+}
+
+variable "environment" {
+}
+
+variable "account_id" {
+}
+
+data "template_file" "policy" {
+  template = "${file("${path.module}/policy.json")}"
+
+  vars = {
+    bucket     = "${var.name}-${var.environment}-transfer"
+    account_id = "${var.account_id}"
+  }
+}
+
+resource "aws_s3_bucket" "transfer" {
```

Comment:
you guys are going to want to have more control over the object versioning, cloudwatch logging and server side encryption

Versioning:
https://www.terraform.io/docs/providers/aws/r/s3_bucket.html#using-versioning

Logging:
https://www.terraform.io/docs/providers/aws/r/s3_bucket.html#enable-logging

Encryption:
https://www.terraform.io/docs/providers/aws/r/s3_bucket.html#enable-default-server-side-encryption